src/app/cli/commands/configure.ts

```diff
@@ -0,0 +1,856 @@
+import * as p from '@clack/prompts';
+import chalk from 'chalk';
+import fs from 'fs/promises';
+import path from 'path';
+import * as YAML from 'yaml';
+import {
+    MCP_SERVER_REGISTRY,
+    getMcpServerCategories,
+    getMcpServersByCategory,
+} from '@core/config/mcp-registry.js';
+import { ConfigurationManager } from '@core/config/config-manager.js';
+import {
+    getDefaultModelForProvider,
+    LLMProvider,
+    logger,
+    getSupportedModels,
+} from '@core/index.js';
+import type { AgentConfig, McpServerConfig } from '@core/config/schemas.js';
+import { getPrimaryApiKeyEnvVar } from '@core/utils/api-key-resolver.js';
+import os from 'os';
+
+export interface ConfigureCommandOptions {
+    /** Whether to save the configuration after creation */
+    save?: boolean;
+    /** Output path for the configuration file */
+    output?: string;
+    /** Load an existing configuration to modify */
+    load?: string | boolean;
+    /** List saved configurations */
+    list?: boolean;
+    /** Delete a saved configuration */
+    delete?: string;
+    /** Export a saved configuration */
+    export?: string;
+    /** Quick path (minimal prompts) */
+    quick?: boolean;
+}
+
+// ────────────────────────── Local MCP registry helpers ─────────────────────────
+const LOCAL_MCP_REGISTRY_PATH = path.join(os.homedir(), '.saiki', 'mcp-registry.local.json');
+
+async function loadLocalMcpRegistry(): Promise<Record<string, any>> {
+    try {
+        const data = await fs.readFile(LOCAL_MCP_REGISTRY_PATH, 'utf-8');
+        return JSON.parse(data);
+    } catch (_err) {
+        return {};
+    }
+}
+
+/**
+ * Main configure command - provides interactive agent configuration
+ */
+export async function configureCommand(options: ConfigureCommandOptions = {}): Promise<void> {
+    const configManager = new ConfigurationManager();
+
+    try {
+        // Handle list command
+        if (options.list) {
+            await listConfigurations(configManager);
+            return;
+        }
+
+        // Handle delete command
+        if (options.delete) {
+            await deleteConfiguration(configManager, options.delete);
+            return;
+        }
+
+        // Handle export command
+        if (options.export) {
+            await exportConfiguration(configManager, options.export, options.output);
+            return;
+        }
+
+        // Handle configuration creation/modification
+        p.intro(chalk.inverse(' Saiki Agent Configuration '));
+
+        let baseConfig: AgentConfig | undefined;
+        let configName: string | undefined;
+        let loadedConfigId: string | undefined;
+
+        // Load existing configuration if specified
+        if (options.load !== undefined) {
```

Comment:
we are checking for `if (options.load !== undefined)` but we can inverse this if to avoid a huge if block and improve readability
```
if (options.load === undefined) { 
    continue; 
}
// Now we don't need the top level if 
if (options.load === true || options.load === '') {
    ...remaining code
}
```


then continue with the rest of the code - removes one nested block

this is a general good practice